src/model/CoilCoolingDXSingleSpeedThermalStorage.cpp

```diff
@@ -0,0 +1,3253 @@
+                                                   "Cooling Coil Evaporative Condenser Pump Electricity Energy",
+                                                   "Cooling Coil Basin Heater Electricity Rate",
+                                                   "Cooling Coil Basin Heater Electricity Energy",
+                                                   "Cooling Coil Thermal Storage Plant Heat Transfer Rate",
+                                                   "Cooling Coil Thermal Storage Plant Heat Transfer Energy",
+                                                   "Cooling Coil Fluid Thermal Storage End Temperature"};
+      return result;
+    }
+
+    IddObjectType CoilCoolingDXSingleSpeedThermalStorage_Impl::iddObjectType() const {
+      return CoilCoolingDXSingleSpeedThermalStorage::iddObjectType();
+    }
+
+    std::vector<ScheduleTypeKey> CoilCoolingDXSingleSpeedThermalStorage_Impl::getScheduleTypeKeys(const Schedule& schedule) const {
+      std::vector<ScheduleTypeKey> result;
+      const UnsignedVector fieldIndices = getSourceIndices(schedule.handle());
+      if (std::find(fieldIndices.cbegin(), fieldIndices.cend(), OS_Coil_Cooling_DX_SingleSpeed_ThermalStorageFields::AvailabilitySchedule)
+          != fieldIndices.cend()) {
+        result.emplace_back("CoilCoolingDXSingleSpeedThermalStorage", "Availability Schedule");
+      }
+      if (std::find(fieldIndices.cbegin(), fieldIndices.cend(), OS_Coil_Cooling_DX_SingleSpeed_ThermalStorageFields::OperationModeControlSchedule)
+          != fieldIndices.cend()) {
+        result.emplace_back("CoilCoolingDXSingleSpeedThermalStorage", "Operation Mode Control Schedule");
+      }
+      if (std::find(fieldIndices.cbegin(), fieldIndices.cend(), OS_Coil_Cooling_DX_SingleSpeed_ThermalStorageFields::BasinHeaterAvailabilitySchedule)
+          != fieldIndices.cend()) {
+        result.emplace_back("CoilCoolingDXSingleSpeedThermalStorage", "Basin Heater Availability Schedule");
+      }
+      return result;
+    }
+
+    unsigned CoilCoolingDXSingleSpeedThermalStorage_Impl::inletPort() const {
+      return OS_Coil_Cooling_DX_SingleSpeed_ThermalStorageFields::EvaporatorAirInletNode;
+    }
+
+    unsigned CoilCoolingDXSingleSpeedThermalStorage_Impl::outletPort() const {
+      return OS_Coil_Cooling_DX_SingleSpeed_ThermalStorageFields::EvaporatorAirOutletNode;
+    }
+
+    ModelObject CoilCoolingDXSingleSpeedThermalStorage_Impl::clone(Model model) const {
+      auto newCoil = StraightComponent_Impl::clone(model).cast<CoilCoolingDXSingleSpeedThermalStorage>();
+
+      return std::move(newCoil);
+    }
+
+    std::vector<ModelObject> CoilCoolingDXSingleSpeedThermalStorage_Impl::children() const {
+      std::vector<ModelObject> result;
+
+      // TODO: every curve?
+
+      return result;
+    }
+
+    boost::optional<HVACComponent> CoilCoolingDXSingleSpeedThermalStorage_Impl::containingHVACComponent() const {
```

Comment:
Do we need a containingZoneHVACComponent?

Reply:
No.